docs/apis/core/customfields/index.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Custom fields API
+tags: []
+---
+
+import { Since } from '@site/src/components';
+
+<Since versions={["3.7"]} issueNumber="MDL-57898" />
+
+## Custom fields API overview
+
+Custom fields API was added in Moodle 3.7. It allows to configure custom fields that can be added to various contexts. Each **component** (or plugin) that wants to use custom fields can define several **areas**. For example, 'core_course' component defines an area 'course' that allows to add custom fields to the courses, the same component can define another area 'coursecat' that will allow to add custom fields to the course categories. Inside each area the component/plugin can decide whether to use or not to use **itemid**. For example, course custom fields are the same throughout the system and they don't use itemid (it is always 0). But there could be an activity module that would want to configure different custom fields for each individual instance of module, then this module would use the module id as the itemid, as in ([example](https://github.com/marinaglancy/moodle-mod_surveybuilder)). This would allow to create modules similar to mod_data and mod_feedback where each instance has it's own set of fields.
```

Comment:
```suggestion
Custom fields API allows to configure custom fields that can be added to various contexts. Each **component** (or plugin) that wants to use custom fields can define several **areas**. 

:::info Example
 
The `core_course` component defines an area `course` that allows to add custom fields to the courses. The same component can define another area `coursecat` that will allow to add custom fields to the course categories.

:::

Inside each area, the component/plugin can decide whether to use or not to use **itemid**.

:::info Example
 
Course custom fields are the same throughout the system and they don't use `itemid` (it is always 0). But there could be an activity module that would want to configure different custom fields for each individual instance of module. Then this module would use the module id as the `itemid`, as in ([example](https://github.com/marinaglancy/moodle-mod_surveybuilder)). This would allow to create modules similar to `mod_data` and `mod_feedback` where each instance has its own set of fields.

:::
```